src/nvme/fabrics.c

```diff
@@ -811,6 +812,23 @@ int nvmf_add_ctrl(nvme_host_t h, nvme_ctrl_t c,
 
 	}
 
+	root_app = nvme_root_get_application(h->r);
+	app = nvme_subsystem_get_application(s);
+	if (root_app) {
+		/*
+		 * configuration is managed by an application,
+		 * refuse to act on subsystems which either have
+		 * no application set or which habe a different
+		 * application string.
+		 */
+		if (!app || strcmp(app, root_app)) {
```

Comment:
For blktests the calling application is nvme-cli itself via udev. So should we start trying to figure in which context nvme-cli is started or should just have a way to say ignore subsys with a given name should be ignored?

Reply:
Basically the problem is how to figure out the calling context for nvme-cli to get this working here.